### PR TITLE
xfail instead of skip

### LINKED ISSUE #12: `DeprecationWarning`: coords should be an ndarray
```
../../../../../opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/sparse/_coo/core.py:261
../../../../../opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/sparse/_coo/core.py:261
../../../../../opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/sparse/_coo/core.py:261
tests/test_sparseconverters.py::test_make_like[sparse.COO-source_array1]
  /opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/sparse/_coo/core.py:261: DeprecationWarning: coords should be an ndarray. This will raise a ValueError in the future.
    warnings.warn(
```

### LINKED ISSUE #44: Instead of `pytest.skip`, maybe use `pytest.xfail` for unsupported combinations?
Here, for instance:

https://github.com/LiberTEM/sparseconverter/blob/cbf593cb8d33c067ffa44ad767d180fc452de6cc/tests/test_sparseconverters.py#L191-L192

Using `pytest.skip` means there is a considerable amount of noise in the output, and it is not easy to figure out if important tests are skipped (for example, when running with vs without `cupy` - if `cupy` doesn't cleanly import for some reason, the important skip drowns in a sea of other test cases and combinations that are _always_ skipped)

### PR DESCRIPTION
**Pull request by uellue**
...and get rid of a deprecation warning.

Closes #45 
Closes #44 
Closes #12

### CODE REVIEW

**Comment by uellue** [comment]
Sry @sk1p I saw your PR only now!

File: src/sparseconverter/__init__.py [hunk 1]
```diff
@@ -340,20 +340,23 @@ def _build_converters(self):
                     self._converters[(left, right)] = _identity
             self._converters[(NUMPY, NUMPY_MATRIX)] = chain(_flatsig, np.asmatrix)
             self._converters[(NUMPY_MATRIX, NUMPY)] = np.asarray
-            # Support direct construction from each other
-            for left in (
-                        NUMPY, CUDA, SPARSE_COO, SPARSE_GCXS, SPARSE_DOK,
-                    ):
+
+            for left in (NUMPY, CUDA, ):
                 for right in SPARSE_COO, SPARSE_GCXS, SPARSE_DOK:
```
Comment (sk1p) [inline src/sparseconverter/__init__.py:345]:
```suggestion
            for left in NUMPY, CUDA:
                for right in SPARSE_COO, SPARSE_GCXS, SPARSE_DOK:
```

File: src/sparseconverter/__init__.py [hunk 2]
```diff
@@ -340,20 +340,23 @@ def _build_converters(self):
...
-            )
+                        self._converters[(left, right)] = _classes[right].from_numpy
+
+            formatcodes = {
+                SPARSE_COO: 'coo',
+                SPARSE_GCXS: 'gcxs',
+                SPARSE_DOK: 'dok',
+            }
+            for left in (SPARSE_COO, SPARSE_GCXS, SPARSE_DOK, ):
+                for right in SPARSE_COO, SPARSE_GCXS, SPARSE_DOK:
```
Comment (sk1p) [inline src/sparseconverter/__init__.py:355]:
```suggestion
            for left in SPARSE_COO, SPARSE_GCXS, SPARSE_DOK:
                for right in SPARSE_COO, SPARSE_GCXS, SPARSE_DOK:
```

File: src/sparseconverter/__init__.py [hunk 3]
```diff
@@ -904,175 +907,175 @@ def get(self, item, default):
 # See scripts/benchmark.ipynb to generate this list
 # FIXME improve cost basis based on real use cases.
 _cost = {
-    (CUDA, CUDA): 1.397e-05,
-    (CUDA, CUPY): 0.00290094,
-    (CUDA, CUPY_SCIPY_COO): 0.03676262,
```
Comment (sk1p) [inline src/sparseconverter/__init__.py:909]: It's quite hard to review the `_cost` changes - did you notice any obvious outliers or large changes to the previous values?

**Review by sk1p — APPROVED** [review]
Generally LGTM! I've added some style suggestions for consistency (parentheses are no longer needed, as the tuples fit into one line), feel free to ignore.

Curious about the `_cost` thing.

File: src/sparseconverter/__init__.py (hunk 3 above)
Comment (uellue) [reply src/sparseconverter/__init__.py:909]: I checked the affected ones, there are no outliers.

**Pull request merged**